File: BillingAndSubscriptionSystem/WebApi/BillingAndSubscriptionSystem.WebApi/Constants/RouteKey.cs
namespace BillingAndSubscriptionSystem.WebApi.Constants
{
    public class RouteKey
    {
        // Base Route
        public const string MainRoute = "api/BillingAndSubscription";

        // Billing Routes
        public const string BillingRoute = MainRoute + "/Billing";
        public const string GetUsersWithBilling = "GetUsersWithBilling";
        public const string UpdateBilling = "UpdateBilling";

        // Payment Routes
        public const string PaymentRoute = MainRoute + "/Payments";
        public const string CreatePayment = "CreatePayment";
        public const string GetOverDuePayment = "GetOverduePayments";

        // Subscription Routes
        public const string UserSubscriptionRoute = MainRoute + "/UserSubscription";
        public const string GetSubscriptionByUserId = "GetSubscriptionByUserId/{userId}";
        public const string CreateUserSubscriptionPlan = "CreateUserSubscriptionPlan";
        public const string UpdateUserSubscriptionPlan = "UpdateUserSubscriptionPlan/{Id}";
        public const string DeleteUserSubscriptionPlan = "DeleteUserSubscriptionPlan/{Id}";

        // User Routes
        public const string UserRoute = MainRoute + "/User";
        public const string GetUsers = "GetUsers";
        public const string GetUserById = "GetUserById/{userId}";
        public const string AddUser = "AddUser";
        public const string UpdateUser = "UpdateUser/{userId}";
        public const string DeleteUser = "DeleteUser/{userId}";

        // Auth Routes
        public const string AuthRoute = MainRoute + "/Auth";
        public const string Login = "Login";
        public const string Register = "Register";

        // Menu Routes
        public const string MenuRoute = MainRoute + "/Menu";
        public const string GetSidebarMenu = "GetSidebarMenu";

        // SubMenu Routes
        public const string SubMenuRoute = MainRoute + "/SubMenu";
        public const string GetAllSubMenu = "GetAllSubMenu";

        // Analytics Routes
        public const string AnalyticsRoute = MainRoute + "/Analytics";
        public const string GetAnalyticsState = "AnalyticsState";
    }
}

File: authProject/src/app/shared/constants/routeKey.ts
export const RouteKey = {
  // AUTHENTICATION ROUTES
  AUTH_LOGIN_URL: 'Auth/login',
  AUTH_REGISTER_URL: 'Auth/register',

  // SIDEBAR MENU ROUTES
  GET_ALL_MENU_ITEMS_URL: 'Menu/GetSidebarMenu',

  // SUBSCRIPTION ROUTES
  GET_SUBSCRIPTION_BY_ID_URL: 'UserSubscription/GetSubscriptionByUserId',
  CREATE_SUBSCRIPTION_URL: 'UserSubscription/CreateUserSubscriptionPlan',
  UPDATE_SUBSCRIPTION_URL: 'UserSubscription/UpdateUserSubscriptionPlan',

  // USER ROUTES
  GET_USER_BY_ID_URL: 'User/GetUserById',
  GET_ALL_USERS_URL: 'User/GetUsers',
  CREATE_USER_URL: 'User/AddUser',
  UPDATE_USER_URL: 'User/UpdateUser',
  DELETE_USER_URL: 'User/DeleteUser',

  // PAYMENT ROUTES
  CREATE_PAYMENT_URL: 'Payments/CreatePayment',
  FETCH_OVERDUE_PAYMENTS_URL: 'Payments/GetOverduePayments',

  // BILLING ROUTES
  UPDATE_BILLING_URL: 'Billing/UpdateBilling',
  GET_USERS_WITH_BILLING: 'Billing/GetUsersWithBilling'
};
